src/channel/channel.js

```diff
@@ -418,6 +418,14 @@ Channel.prototype.acceptUser = function (user) {
 
         this.logger.log("[login] Accepted connection from Tor exit at " +
                         user.displayip);
+    } 
+    else if(user.isAnonymous()){
+        var anonymousBanned = this.modules.options && this.modules.options.get("block_anonymous_users");
+        if (anonymousBanned) {
+            user.kick("This channel has banned anonymous users. Please register if you'd like to join.");
```

Comment:
This would also block guests (unless they manage to type a guest name before the join callback fires) -- I feel like this should operate similar to the channel password feature and ask anonymous users to input a name before joining.

There might also be a case where people want to block guest names I guess, but I feel like that particular feature might be better solved by a whitelist (since registration doesn't require an email address or anything, "registered user" doesn't really have any more legitimacy than a guest name).

Reply:
Pardon the comment spam :) I realized I forgot to continue as a reply. 

I think there are probably three different feature possibilities here. One feature, block all guests. Another feature, block all guests without a guest name. And finally, a third feature, having a white list.

While they are different, I think maybe the first two do cover a lot of the same ground. I obviously liked the more broad blocking of all guests as a feature, but if you prefer blocking of guests with no name assigned, that's a really easy update. Edit: It's possible, too, these two ideas could co-exist, with the allowing of guests with names being an setting the channel owner could toggle.

Either way, I would say the white list definitely seems like a nice feature that would exist in addition to one, or both of those previously features. While a white list could provide the feature of disallowing guests, I think it's appreciably different than either flavours of blocking guests in its scope.

Reply:
And just to avoid all risks of ideas getting twisted up because I'm using vocab that I'm not exactly rock solid on for this project - the intent of the original pull requests was to block all non-registered users from a channel, which I refereed to as anonymous users, even those who might have guest login named assigned.   In my reply to your response, when I'm referring to 'guests', I just mean any non-registered user.

Reply:
In terms of the code, "anonymous" and "guest" are separate ranks with subtly different permissions; anonymous users have no name and hence do not appear in the userlist (unless they choose to enter a guest name later, in which case they become a guest).

Supporting blocking either one as options would be a possibility; I guess it depends on what your motivation is for using the feature.  Maybe I am missing some context on why the distinction between registered and guest name is important.  My goal here is to find the right balance between giving channel owners control of their channel and minimizing frustration and annoyance for users -- unless the channel has good reason to block guest names, I think "please enter a name to join" would be less annoying than "please register an account".

This also raises a question of whether channels which choose to enable this restriction should be disallowed from also setting the channel to "public" (i.e., list it on the index page).  It's a pretty annoying experience to browse the index page, click on something interesting, and immediately get a "You must register to continue" prompt.  It seems to me like this feature is mostly useful for private channels wishing to have a higher degree of accountability for who is joining.

Agree that whitelist is a separate feature.

Reply:
Sweet, thanks for clearing up the lingo :) So, uh, I'm convinced by your arguments. There is a sort of tighter relationship to a user and a registered account, but like you said, the barrier to entry for registering is very low., so it's probably not that important  To that end, I do think disallowing anonymous connections, but allowing guest logins, satisfies what I would hope to have from the feature as a channel owner, which is just having more visibility with regards to who is in the channel.  I definitely agree that it's certainly a more user friendly experience for the anonymous / guest user, so that seems good, too.  Would have your blessing to work that out?  

As far as my two cents regarding the question of whether or not to list channels publicly with this feature..to me, I think that only comes into play when considering the block guests + anonymous users feature,  In that case, the behaviour I would expect would be for these channels to not be listed for guest / anonymous accounts.  If the channel setting was just blocking anonymous u, but allow guest logins, I think listing publicly could be O.K. provided we give the anonymous user an avenue go provide a guest name.

Reply:
That seems reasonable.